dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DefaultUploadSchedulerStrategy.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.upload
+
+import com.datadog.android.core.configuration.UploadSchedulerStrategy
+import com.datadog.android.core.internal.configuration.DataUploadConfiguration
+import com.datadog.android.core.internal.data.upload.DataOkHttpUploader.Companion.HTTP_ACCEPTED
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToLong
+
+internal class DefaultUploadSchedulerStrategy(
+    internal val uploadConfiguration: DataUploadConfiguration
+) : UploadSchedulerStrategy {
+
+    private val currentDelays = mutableMapOf<String, Long>()
```

Comment:
We probably need to make this one thread-safe, just in case. Even though we have only a single upload thread currently, it may change in the future.

Reply:
Good point